src/LibraryManager.Vsix/Commands/InstallLibraryCommand.cs

```diff
@@ -60,52 +59,67 @@ public static void Initialize(Package package, OleMenuCommandService commandServ
             button.Visible = button.Enabled = false;
 
             ProjectItem item = await VsHelpers.GetSelectedItemAsync();
+            Project project = await VsHelpers.GetProjectOfSelectedItemAsync().ConfigureAwait(false);
 
-            if (item?.ContainingProject == null)
+            if (item?.ContainingProject == null && project == null)
             {
                 return;
             }
 
-            if (VSConstants.ItemTypeGuid.PhysicalFolder_string.Equals(item.Kind, StringComparison.OrdinalIgnoreCase))
-            {
-                button.Visible = true;
-                button.Enabled = KnownUIContexts.SolutionExistsAndNotBuildingAndNotDebuggingContext.IsActive && !_libraryCommandService.IsOperationInProgress;
-            }
+            button.Visible = true;
+            button.Enabled = KnownUIContexts.SolutionExistsAndNotBuildingAndNotDebuggingContext.IsActive && !_libraryCommandService.IsOperationInProgress;
         }
 
         private async Task ExecuteAsync(object sender, EventArgs e)
         {
             Telemetry.TrackUserTask("installdialogopened");
 
             ProjectItem item = await VsHelpers.GetSelectedItemAsync().ConfigureAwait(false);
+            Project project = await VsHelpers.GetProjectOfSelectedItemAsync().ConfigureAwait(false);
 
-            if (item != null)
+            if (project != null)
             {
-                string target = item.FileNames[1];
+                string target = string.Empty;
+                string rootFolder = await project.GetRootFolderAsync().ConfigureAwait(false);
 
-                Project project = await VsHelpers.GetProjectOfSelectedItemAsync().ConfigureAwait(false);
-
-                if (project != null)
+                // Install command was invoked from a folder.
+                // So the initial target location should be name of the folder from which
+                // the command was invoked.
+                if (item != null)
                 {
-                    string rootFolder = await project.GetRootFolderAsync().ConfigureAwait(false);
+                    target = item.FileNames[1];
```

Comment:
>item.FileNames[1] [](start = 29, length = 17)

This includes the separator?

Reply:
Yes, this already includes the separator

---
In reply to: [201047703](https://github.com/aspnet/LibraryManager/pull/291#discussion_r201047703) [](ancestors = 201047703)